onnxruntime/test/providers/kernel_def_hash_test.cc

```diff
@@ -0,0 +1,810 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/**
+ * IMPORTANT NOTE AT THE TOP OF THE FILE
+ *
+ * This file contains tests which verify expected kernel def hashes.
+ * It is important for these to remain stable so that ORT format models are
+ * backward compatible.
+ *
+ * If you are seeing a test failure from one of these tests, it is likely that
+ * some kernel definition changed in a way that updated its hash value.
+ * This is what we want to catch! Please update the kernel definition.
+ * If adding more supported types to an existing kernel definition, consider
+ * using KernelDefBuilder::FixedTypeConstraintForHash().
+ *
```

Comment:
Maybe add an example of what the before/after registration would look like if we were adding a type to the constraints and needed to use the fixed type constraint value.

May also be good to mention that any FixedTypeConstraintForHash usage should not carry over to a new registration. Possible someone could cut-and-paste and not realize that a new registration should start with a clean setup.